packages/toolkit/src/amazonqGumby/activation.ts

```diff
@@ -13,14 +13,43 @@ import { transformByQState } from '../codewhisperer/models/model'
 import * as CodeWhispererConstants from '../codewhisperer/models/constants'
 import { ProposedTransformationExplorer } from '../codewhisperer/service/transformationResultsViewProvider'
 import { codeTransformTelemetryState } from './telemetry/codeTransformTelemetryState'
-import { telemetry } from '../shared/telemetry/telemetry'
-import { CancelActionPositions, logCodeTransformInitiatedMetric } from './telemetry/codeTransformTelemetry'
+import { CodeTransformJavaSourceVersionsAllowed, telemetry } from '../shared/telemetry/telemetry'
+import {
+    CancelActionPositions,
+    JDKToTelemetryValue,
+    logCodeTransformInitiatedMetric,
+} from './telemetry/codeTransformTelemetry'
 import { CodeTransformConstants } from './models/constants'
+import { getOpenProjects, validateOpenProjects } from '../codewhisperer/service/transformByQHandler'
+import { MetadataResult } from '../shared/telemetry/telemetryClient'
 
 export async function activate(context: ExtContext) {
     const transformationHubViewProvider = new TransformationHubViewProvider()
     new ProposedTransformationExplorer(context.extensionContext)
 
+    // Register an activation event listener to determine when the IDE opens, closes or users
+    // select to open a new workspace
+    const workspaceChangeEvent = vscode.workspace.onDidChangeWorkspaceFolders(event => {
```

Comment:
Is there a way to avoid setting this up if gumby has not been used? (Same general question for all gumby features, but that's a backlog item...)